### PR TITLE
Regularly try to wake up dests instead of waiting for next PDU/EDU

### PR DESCRIPTION
**Pull request by MatMaul**
Currently if we fail to send a transaction, we never retry it until another PDU/EDU is available to send to this destination.

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog).
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: synapse/federation/sender/__init__.py [hunk 1]
```diff
@@ -184,14 +183,18 @@
...
-# Time (in s) after Synapse's startup that we will begin to wake up destinations
-# that have catch-up outstanding.
-CATCH_UP_STARTUP_DELAY_SEC = 15
+# Time (in s) to wait before trying to wake up destinations that have
+# catch-up outstanding. This will also be the delay applied at startup
+# before trying the same.
+# Please note that rate limiting still applies, so while the loop is
+# executed every X seconds the destinations may not be wake up because
+# they are being rate limited following previous attempt failures.
+WAKEUP_RETRY_PERIOD_SEC = 60
```
Comment (MadLittleMods) [inline synapse/federation/sender/__init__.py:192]: We also need to update the big description at the top of this file on how all of this works now:

https://github.com/matrix-org/synapse/blob/f6321e386c804017f04c7bdfddda6afb618702ba/synapse/federation/sender/__init__.py#L108-L125

Comment (MatMaul) [reply synapse/federation/sender/__init__.py:190]: Good point, thanks !
Done.

File: synapse/federation/sender/__init__.py [hunk 2]
```diff
@@ -415,12 +416,10 @@ def __init__(self, hs: "HomeServer"):
             / hs.config.ratelimiting.federation_rr_transactions_per_room_per_second
         )
 
-        # wake up destinations that have outstanding PDUs to be caught up
-        self._catchup_after_startup_timer: Optional[
-            IDelayedCall
-        ] = self.clock.call_later(
-            CATCH_UP_STARTUP_DELAY_SEC,
+        # Regularly wake up destinations that have outstanding PDUs to be caught up
+        self.clock.looping_call(
```
Comment (MadLittleMods) [inline synapse/federation/sender/__init__.py:420]: Do we have to worry about multiple `_wake_destinations_needing_catchup()` calls stacking up if they don't finish within the hour?

Does `looping_call` handle that de-duplication for us? (I don't see anything in the docstring)

File: tests/federation/test_federation_catch_up.py [hunk 3]
```diff
@@ -431,28 +431,22 @@ def test_catch_up_on_synapse_startup(self) -> None:
...
-        # cancel the pre-existing timer for _wake_destinations_needing_catchup
-        # this is because we are calling it manually rather than waiting for it
-        # to be called automatically
-        assert self.federation_sender._catchup_after_startup_timer is not None
-        self.federation_sender._catchup_after_startup_timer.cancel()
-
-        self.get_success(
-            self.federation_sender._wake_destinations_needing_catchup(), by=5.0
-        )
+        self.pump(by=5.0)
```
Comment (MadLittleMods) [inline tests/federation/test_federation_catch_up.py:441]: Does this magic value pertain to anything? We should comment why we wait this long

I'm guessing this is something to do with `WAKEUP_INTERVAL_BETWEEN_DESTINATIONS_SEC` but I would expect to pump longer in order to wake up all 26 destinations.

**Review by MadLittleMods — APPROVED** [review]

File: tests/federation/test_federation_catch_up.py (hunk 3 above)
Comment (MatMaul) [reply tests/federation/test_federation_catch_up.py:441]: I am guessing it's related to this `* 100` here: https://github.com/matrix-org/synapse/blob/develop/tests/unittest.py#L619.

I am going to add a comment.

**Pull request merged**